sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePlugin.scala

```diff
@@ -31,52 +31,54 @@ object ScalaNativePlugin extends AutoPlugin {
     }
 
     val nativeConfig =
-      taskKey[build.NativeConfig](
+      settingKey[build.NativeConfig](
         "User configuration for the native build, NativeConfig"
       )
 
+    val nativeLink =
+      taskKey[File]("Generates native binary without running it.")
+
+    // 10 raw settings for minimal sbt projects
+
     val nativeClang =
```

Comment:
All of the settings below were assumed to be deprecated and replaced with NativeConfig. Removing them would significantly simplify the whole initialization - we no longer need to generate nativeConfig, assign parts of it's to settingKeys with default value, and finally read these settings to populate a final nativeConfig

Reply:
Don't we need them for `sbt` synthesized root project settings?

```scala
logLevel := Level.Debug

nativeMode := "release-fast"

nativeLinkStubs := true
```

Reply:
Yes, I see that you can use the following for example in a  basic `build.sbt` file.

```scala
import scala.scalanative.build._

nativeConfig ~= {
    _.withMode(Mode.releaseFast)
}
```

Reply:
Removing raw setting breaks some scripted tests so that is in the works.